eclair-core/src/test/scala/fr/acinq/eclair/wire/LightningMessageCodecsSpec.scala

```diff
@@ -58,6 +59,53 @@ class LightningMessageCodecsSpec extends FunSuite {
     }
   }
 
+  test("encode/decode with uint64L codec") {
+    val expected = Map(
+      0L -> hex"00 00 00 00 00 00 00 00",
+      42L -> hex"2a 00 00 00 00 00 00 00",
+      6211610197754262546L -> hex"12 34 56 78 90 12 34 56"
+    ).mapValues(_.toBitVector)
+
+    for ((long, ref) <- expected) {
+      val encoded = uint64L.encode(long).require
+      assert(ref === encoded)
+      val decoded = uint64L.decode(encoded).require.value
+      assert(long === decoded)
+    }
+  }
+
+  test("encode/decode with varint codec") {
```

Comment:
Does Bitcoin Core have test vectors for the varint encoding?

Reply:
Good idea, I'll find them.

Reply:
It looks like I actually came up with almost the same test suite as they did ;)
Adding a few more for completeness.